AutoReplayUploader/AutoReplayUploaderPlugin.cpp

```diff
@@ -5,194 +5,341 @@
 	{
 		cvarManager->log("Could not find all functions in SteamAPI DLL!");
 		return;
 	}
 
-	gameWrapper->HookEventWithCaller<ServerWrapper>("Function TAGame.GameEvent_Soccar_TA.EventMatchEnded", 
-		std::bind(&AutoReplayUploaderPlugin::OnGameComplete, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3));
-
-	//cvarManager->registerCvar("cl_autoreplayupload_filepath", "./bakkesmod/data/autoreplaysave.replay", "Path to save to be uploaded replay to.");
-	cvarManager->registerCvar("cl_autoreplayupload_calculated", "0", "Upload to replays to calculated.gg automatically", true, true, 0, true, 1).bindTo(uploadToCalculated);
-
-	cvarManager->registerCvar("cl_autoreplayupload_ballchasing", "0", "Upload to replays to ballchasing.com automatically", true, true, 0, true, 1).bindTo(uploadToBallchasing);
-	
-
-	//Auth token response, stored in cvar so we can display it in the plugins tab. Should not be exposed to user!
-	cvarManager->registerCvar("cl_autoreplayupload_ballchasing_testkeyresult", "Untested", "Auth token needed to upload replays to ballchasing.com", false, false, 0, false, 0, false);
-	cvarManager->registerCvar("cl_autoreplayupload_ballchasing_visibility", "public", "Replay visibility when uploading to ballchasing.com", false, false, 0, false, 0, false).addOnValueChanged([this](std::string oldValue, CVarWrapper cvar)
-	{
-		//cvarManager->log(GenerateUrl(BALLCHASING_ENDPOINT_DEFAULT, { { "visibility", cvar.getStringValue() } }));
-	});;
-	cvarManager->registerCvar("cl_autoreplayupload_ballchasing_authkey", "", "Auth token needed to upload replays to ballchasing.com").addOnValueChanged([this](std::string oldVal, CVarWrapper cvar)
-	{
-		//User changed authkey, reset testkeyresult
-		cvarManager->getCvar("cl_autoreplayupload_ballchasing_testkeyresult").setValue("Untested");
-	});
-	cvarManager->registerNotifier("cl_autoreplayupload_ballchasing_testkey", std::bind(&AutoReplayUploaderPlugin::TestBallchasingAuth, this, std::placeholders::_1), 
-		"Checks whether ballchasing authkey is valid", PERMISSION_ALL);
-	//cvarManager->registerCvar("cl_autoreplayupload_calculated_endpoint", CALCULATED_ENDPOINT_DEFAULT, "URL to upload replay to when uploading to calculated.gg instance");
-
-	/*
-	Load notification assets
-	*/
-	gameWrapper->LoadToastTexture("calculated_logo", "./bakkesmod/data/assets/calculated_logo.tga");
-	gameWrapper->LoadToastTexture("ballchasing_logo", "./bakkesmod/data/assets/ballchasing_logo.tga");
-	cvarManager->registerCvar("cl_autoreplayupload_notifications", "1", "Show notifications on successful uploads", true, true, 0, true, 1).bindTo(showNotifications);
-
+	steamUserName = steamClient->GetISteamFriends(steamUser, steamPipe, "SteamFriends017")->GetPersonaName();
 }
 
+/**
+* OnUnload event called when the plugin is unloaded by BakkesMod
+*/
 void AutoReplayUploaderPlugin::onUnload()
 {
```

Comment:
Do we need to do anything so cvar variables persist across loads of bakkes mod?

Reply:
By default, cvars only get saved when the user closes the F2 menu or when the "writeconfig" command is executed. You could add a change notifier to the cvar by using `addOnValueChanged` on a cvarwrapper, which executes `cvarManager->executeCommand("writeconfig");`.

onUnload is not guaranteed to be called when the user simply exists the process, it'll only be guaranteed to be called when the user uses `plugin unload xxx`. Maybe the destructor will get called, but I'm not 100% sure about that. The method I described above is probably your best bet if you want a persistent number tracker.